File: src/java/org/javaturk/dp/ch04/abstractFactory/university/sol3/FactoryForUniversityA.java
package org.javaturk.dp.ch04.abstractFactory.university.sol3;

public interface FactoryForUniversityA {
	
	Course createCourse(String name);
	Professor createProfessor(String name);
	Schedule createSchedule(String name);
	Syllabus createSyllabus(String name);
}

File: src/typescript/com/gokselkucuksahin/dp/creational-patterns/abstract-factory/university/sol3/factory-for-university-a.ts
import {Course} from "../sol1/course";
import {Professor} from "../sol1/professor";
import {Schedule} from "../sol1/schedule";
import {Syllabus} from "../sol1/syllabus";

export interface FactoryForUniversityA {
  createCourse(name: string): Course;

  createProfessor(name: string): Professor;

  createSchedule(name: string): Schedule;

  createSyllabus(name: string): Syllabus;
}